osgp/platform/osgp-adapter-ws-core/src/main/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementService.java

```diff
@@ -186,7 +186,8 @@ public Organisation findOrganisation(@Identification final String organisationId
     @Transactional(value = "transactionManager")
     public Page<Event> findEvents(@Identification final String organisationIdentification,
             final String deviceIdentification, final PageSpecifier pageSpecifier, final DateTime from,
-            final DateTime until, final List<EventType> eventTypes) throws FunctionalException {
+            final DateTime until, final List<EventType> eventTypes, final String description,
+            final String descriptionStartsWith) throws FunctionalException {
```

Comment:
We now supply both "description" en "descriptionStartsWith". The code uses an "or" operator for those fields. This isn't obvious to the users. To keep it simple and clear, we could offer just the "descriptionStartsWith". We can probably live without the exact match.

Another option would be to use wildcards, but that would be extending the scope of the story. It would extend the scope, because to be consistent you would have to support wildcards for the deviceIdentification (and possibly the organisationIdentification) as well.

Reply:
this destroys the whole idea that we could do with one call